PiwikTracker/PiwikTracker.swift

```diff
@@ -22,6 +22,8 @@ final public class PiwikTracker: NSObject {
     private var queue: Queue
     internal let siteId: String
     
+    public var logger: Logger? = DefaultLogger(minLevel: .warning)
```

Comment:
I still suggest making the logger non optional even if exposing it as a property.
The code will become more easy to write and reason if you can avoid optional were possible.

You can provide a default implementation of the ´NoOpLogger´ that user can set to make it easy for them.